### PR TITLE
Webp: Media classes

### PR DESCRIPTION
**Pull request by Screenfeed**
See the 4 last commits.

### CODE REVIEW

File: classes/Media/AbstractMedia.php [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+	 * @author Grégory Viguier
+	 *
+	 * @return array
+	 */
+	public function get_allowed_mime_types() {
+		return imagify_get_mime_types( $this->get_context_instance()->get_allowed_mime_types() );
+	}
+
+	/**
+	 * If the media is an image, update the dimensions in the database with the current file dimentions.
```
Comment (Tabrisrp) [inline classes/Media/AbstractMedia.php]:
```suggestion
       * If the media is an image, update the dimensions in the database with the current file dimensions.
```

File: classes/Media/AbstractMedia.php [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+	public function update_dimensions() {
+		if ( ! $this->is_image() ) {
+			// The media is not a supported image.
+			return false;
+		}
+
+		$dimensions = $this->filesystem->get_image_size( $this->get_raw_original_path() );
+
+		if ( ! $dimensions ) {
+			// Could not get the new dimentions.
```
Comment (Tabrisrp) [inline classes/Media/AbstractMedia.php]:
```suggestion
	                // Could not get the new dimensions.
```

File: classes/Media/CustomFolders.php [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+			return [];
+		}
+
+		$original_path = $this->get_raw_original_path();
+
+		if ( ! $original_path ) {
+			return [];
+		}
+
+		$dimentions = $this->get_dimensions();
```
Comment (Tabrisrp) [inline classes/Media/CustomFolders.php]:
```suggestion
		$dimensions = $this->get_dimensions();
```

File: classes/Media/CustomFolders.php [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+		$sizes      = [
+			'full' => [
+				'size'      => 'full',
+				'path'      => $original_path,
+				'width'     => $dimentions['width'],
```
Comment (Tabrisrp) [inline classes/Media/CustomFolders.php]:
```suggestion
				'width'     => $dimensions['width'],
```

File: classes/Media/CustomFolders.php [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+				'height'    => $dimentions['height'],
```
Comment (Tabrisrp) [inline classes/Media/CustomFolders.php]:
```suggestion
				'height'    => $dimensions['height'],
```

File: classes/Media/WP.php [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+			return [];
+		}
+
+		$original_path = $this->get_raw_original_path();
+
+		if ( ! $original_path ) {
+			return [];
+		}
+
+		$dimentions = $this->get_dimensions();
```
Comment (Tabrisrp) [inline classes/Media/WP.php]:
```suggestion
		$dimensions = $this->get_dimensions();
```

File: classes/Media/WP.php [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+		$all_sizes  = [
+			'full' => [
+				'size'      => 'full',
+				'path'      => $original_path,
+				'width'     => $dimentions['width'],
```
Comment (Tabrisrp) [inline classes/Media/WP.php]:
```suggestion
				'width'     => $dimensions['width'],
```

File: classes/Media/WP.php [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+				'height'    => $dimentions['height'],
```
Comment (Tabrisrp) [inline classes/Media/WP.php]:
```suggestion
				'height'    => $dimensions['height'],
```

File: classes/Media/AbstractMedia.php (hunk 1 above)
Comment (Screenfeed) [reply classes/Media/AbstractMedia.php]: Typo, I always do the error. Thanks.

**Pull request merged**